packages/web-react/src/components/Button/useButtonAriaProps.tsx

```diff
@@ -0,0 +1,54 @@
+import { AnchorHTMLAttributes, ButtonHTMLAttributes, ElementType, HTMLAttributes } from 'react';
+import { ClickEvent, AriaButtonProps } from '../../types';
+
+export interface ButtonAria<T> {
+  /** Props for the button element. */
+  buttonProps: T;
+}
+
+export function useButtonAriaProps(props: AriaButtonProps<'a'>): ButtonAria<AnchorHTMLAttributes<HTMLAnchorElement>>;
+export function useButtonAriaProps(
+  props: AriaButtonProps<'button'>,
+): ButtonAria<ButtonHTMLAttributes<HTMLButtonElement>>;
+export function useButtonAriaProps(props: AriaButtonProps<ElementType>): ButtonAria<HTMLAttributes<HTMLElement>>;
+
+export function useButtonAriaProps(props: AriaButtonProps<ElementType>): ButtonAria<HTMLAttributes<any>> {
```

Comment:
This is Typescript Function Overloading, see https://www.tutorialsteacher.com/typescript/function-overloading#:~:text=TypeScript%20provides%20the%20concept%20of,parameters%20should%20be%20the%20same.

The problem is caused by different HTML element, eg. `button` vs `a`, because they have simply different attributes and I must deal with it.